cunumeric/array.py

```diff
@@ -2185,6 +2185,98 @@ def dumps(self):
         """
         return self.__array__().dumps()
 
+    def _normalize_axes_shape(self, axes, s):
+        user_axes  = axes is not None
+        user_sizes = s is not None
+        if user_axes and user_sizes and len(axes) != len(s):
+            raise ValueError(
+                "Shape and axes have different lengths"
+                )
+        if user_axes:
+            fft_axes = axes
+        else:
+            fft_axes = range(-len(s),0) if user_sizes else range(self.ndim)
+        if np.max(np.abs(list(fft_axes))) > self.ndim:
```

Comment:
Can the user provide `axis = [1, 2, 3]` for a 3D fft ? In that case `np.max(np.abs([-3,-2-1]))` will be `3 <= self.ndim`. But it should probably error out? It might be best to keep `axis >= 0`, otherwise the logic for negative/positive axis is a bit annoying to deal with imho.

Reply:
Hmmm I'm not sure what you mean.

axis = [1, 2, 3] would error our correctly for a 3D FFT, it should be axis = [0, 1, 2] to do an FFT over the 3 axes.
And negative axes need to be supported to be consistent with numpy.